src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java

```diff
@@ -487,17 +488,17 @@ public void setDisplayStyle(String selector, String style) {
   }
 
   private void purge(String url) throws Exception {
-    HttpClient client = new HttpClient();
-    HttpMethod method = new PurgeMethod(url);
+    CloseableHttpClient client = HttpClients.createDefault();
+    HttpUriRequest method = new PurgeMethod(url);
     try {
-      int status = client.executeMethod(method);
+      int status = client.execute(method).getStatusLine().getStatusCode();
       if (status != HttpStatus.SC_OK && status != HttpStatus.SC_NOT_FOUND) {
         throw new Exception("HTTP PURGE failed for: " + url + "(" + status + ")");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)